### ISSUE TITLE
Add min/max/clamp to stdlib math

### ISSUE THREAD

**Issue by Temujin2887** [dbaumgarten/yodk#91]
It would be really nice if stdlib/math came with min/max/clamp macros. 

Here's my example implementation, although smarter minds may have more efficient methods:
```
macro math_max(x, y) expr
    x + (y-x)*(x<y)
end

macro math_min(x, y) expr
    x + (y-x)*(x>y)
end

macro math_clamp(x,min,max) expr
    math_min(math_max(x, min), max)
end
```

**Comment by Temujin2887** [comment]
Possibly also a way to remap linear ranges

example implementation:
-  Note: cases where other the old range has a zero size will cause a division by zero. Safety is left off in favor of minimizing the code. 
```
macro math_map(x, old_min, old_max, new_min, new_max) expr
    new_min + (x-old_min) * (new_max-new_min) / (old_max-old_min)
end
```

**Comment by dbaumgarten** [comment]
Min max and clamp is definitely something that should be added to the standard-library. And the code you provided does seem fine to me. I don't really see how it could be written any shorter.

Regsrding map I am not that sure. I don't think I have ssen something like this in any standard-library before and I didn't really miss it anywhere.

**Comment by dbaumgarten** [comment]
Released as part of v0.1.8
